hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestColumnStatsIndex.scala

```diff
@@ -63,6 +69,91 @@ class TestColumnStatsIndex extends HoodieClientTestBase {
     cleanupSparkContexts()
   }
 
+  @Test
+  def testMetadataColumnStatsIndex(): Unit = {
```

Comment:
In general, i think we should test against all operations that can modify Col Stats index (commit, delta_commit, clean, compact, cluster) to make sure they leave it in consistent state.

Reply:
There is already a test for metadata index with table operations in `TestHoodieBackedMetadata#testTableOperationsWithMetadataIndex`

Reply:
Do we assert Column Stats correctness there?

Reply:
Somehow can't resolve my own comment. This is taken care of